src/components/banner/banner.ts

```diff
@@ -0,0 +1,148 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { type Ref, createRef, ref } from 'lit/directives/ref.js';
+
+import { addAnimationController } from '../../animations/player.js';
+import { growVerIn, growVerOut } from '../../animations/presets/grow/index.js';
+import { themes } from '../../theming/theming-decorator';
+import IgcButtonComponent from '../button/button.js';
+import { registerComponent } from '../common/definitions/register.js';
+import type { Constructor } from '../common/mixins/constructor.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { styles } from './themes/banner.base.css.js';
+import { all } from './themes/themes.js';
+
+export interface IgcBannerComponentEventMap {
+  igcClosing: CustomEvent<void>;
+  igcClosed: CustomEvent<void>;
+}
+
+/**
+ * The `igc-banner` component displays important and concise message(s) for a user to address, that is specific to a page or feature.
+ *
+ * @element igc-banner
+ *
+ * @slot - Renders the text content of the banner message.
+ * @slot prefix - Renders additional content at the start of the message block.
+ * @slot actions - Renders any action elements.
+ *
+ * @fires igcClosing - Emitted before closing the banner - when a user interacts (click) with the default action of the banner.
+ * @fires igcClosed - Emitted after the banner is closed - when a user interacts (click) with the default action of the banner.
+ *
+ * @csspart base - The base wrapper of the banner component.
+ * @csspart spacer - The inner wrapper that sets the space around the banner.
+ * @csspart message - The part that holds the text and the illustration.
+ * @csspart illustration - The part that holds the banner icon/illustration.
+ * @csspart content - The part that holds the banner text content.
+ * @csspart actions - The part that holds the banner action buttons.
+ */
+
+@themes(all)
+export default class IgcBannerComponent extends EventEmitterMixin<
+  IgcBannerComponentEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-banner';
+  public static styles = [styles];
+
+  public static register() {
+    registerComponent(IgcBannerComponent, IgcButtonComponent);
+  }
```

Comment:
```suggestion
  /* blazorSuppress */
  public static register() {
    registerComponent(IgcBannerComponent, IgcButtonComponent);
  }
```
Got to make the blazor code generation happy.